addons/point_of_sale/models/pos_session.py

```diff
@@ -667,7 +668,7 @@ def _prepare_line(self, order_line):
         """
         def get_income_account(order_line):
             product = order_line.product_id
-            income_account = product.with_company(order_line.company_id)._get_product_accounts()['income']
```

Comment:
Why do you want to remove this? It is wrong

Reply:
Be cause i put the with_company in a more global place, which means that get_income_account will always be called with the right company already.

Reply:
please don't change stuff like this in point_of_sale module without reasons, it doesn't give any improvement, and doesn't fix anything.